frontend/src/pages/EventCreationPage/components/EventCreationForm/index.tsx

```diff
@@ -29,9 +29,29 @@ interface InitialValuesType {
   hosts: OfficerNameData[];
 }
 
+function findNearestHour(date, isStartDate) {
+  let hour = date.getHours();
+  let startMinutes = date.getMinutes();
+  if (startMinutes === 0) {
+    startMinutes = '00';
+  } else if (startMinutes <= 30) {
+    startMinutes = '30';
+  } else if (startMinutes > 30) {
+    startMinutes = '00';
+    hour += 1;
+  }
+
+  date.setMinutes(startMinutes);
```

Comment:
Please use [setMinutes by date-fns](https://date-fns.org/v2.21.1/docs/setMinutes) instead. I think you're using JS's Date API here? In the future, try to use date-fns as much as you can (you'll most likely never need to explicitly use Date API, but you might need to check its docs once in a while).